Packet++/header/NtpLayer.h

```diff
@@ -0,0 +1,615 @@
+            int8_t pollInterval,
+            /// 8-bit signed integer representing the precision of the system clock, in log2 seconds.
+            precision;
+            /// Total round-trip delay to the reference clock, in NTP short format.
+            uint32_t rootDelay,
+            /// Total dispersion to the reference clock, in NTP short format.
+            rootDispersion,
+            /// 32-bit code identifying the particular server or reference clock.  The interpretation depends on the value in the stratum field.
+            referenceIdentifier;
+            /// Time when the system clock was last set or corrected, in NTP timestamp format.
+            uint64_t referenceTimestamp,
+            /// Time at the client when the request departed for the server, in NTP timestamp format.
+            originTimestamp,
+            /// Time at the client when the request departed for the server, in NTP timestamp format.
+            receiveTimestamp,
+            /// Time at the server when the response left for the client, in NTP timestamp format.
+            transmitTimestamp;
+        };
+#pragma pack(pop)
+
+#pragma pack(push, 1)
+        struct ntp_v3_auth
+        {
+            /// An integer identifying the cryptographic key used to generate the message-authentication code
+            uint32_t keyID;
+            /// This is an integer identifying the cryptographic key used to generate the message-authentication code.
+            uint8_t dgst[8]; // 64 bit DES based
+        };
+#pragma pack(pop)
+
+#pragma pack(push, 1)
+        struct ntp_v4_auth_md5
+        {
+            /// 32-bit unsigned integer used by the client and server to designate a secret 128-bit MD5 key.
+            uint32_t keyID;
+            /// 128-bit MD5 hash
+            uint8_t dgst[16];
+        };
+#pragma pack(pop)
+
+#pragma pack(push, 1)
+        struct ntp_v4_auth_sha1
+        {
+            /// 32-bit unsigned integer used by the client and server to designate a secret 160-bit SHA1 key.
+            uint32_t keyID;
+            /// 160-bit SHA1 hash
+            uint8_t dgst[20];
+        };
+
+        ntp_header *getNtpHeader() const { return (ntp_header *)m_Data; }
+
+    public:
+        /**
+        * Warning of an impending leap second to be inserted or deleted in the last minute of the current month
+        */
+        enum LeapIndicator
+        {
+            NoWarning = 0,
+            /// Last minute of the day has 61 seconds
+            Last61Secs,
+            /// Last minute of the day has 59 seconds
+            Last59Secs,
+            /// Unknown (clock unsynchronized)
+            Unknown
+        };
+
+        /**
+         * Representing the NTP association modes
+         */
+        enum Mode
+        {
+            Reserved = 0,
+            SymActive,
+            SymPassive,
+            Client,
+            Server,
+            Broadcast,
+            Control,
+            PrivateUse
```

Comment:
These enum values are public and not documented, hence Doxygen can't document them:
![image](https://user-images.githubusercontent.com/9059541/151922200-d9d85ec1-9fb4-47aa-ba0d-17d513373c98.png)